src/azure-cli/azure/cli/command_modules/cosmosdb/custom.py

```diff
@@ -319,6 +324,139 @@ def cli_cosmosdb_sql_container_update(client,
+                                      account_name,
+                                      database_name,
+                                      container_name,
+                                      stored_procedure_name,
+                                      stored_procedure_body):
+
+    """Creates or Updates an Azure Cosmos DB SQL stored procedure """
+    sql_stored_procedure_resource = SqlStoredProcedureResource(id=stored_procedure_name)
+    sql_stored_procedure_resource.body = stored_procedure_body
+
+    sql_stored_procedure_create_update_resource = SqlStoredProcedureCreateUpdateParameters(
+        resource=sql_stored_procedure_resource,
+        options={})
+
+    return client.create_update_sql_stored_procedure(resource_group_name,
+                                              account_name,
+                                              database_name,
+                                              container_name,
+                                              stored_procedure_name,
+                                              sql_stored_procedure_create_update_resource)
+
+def cli_cosmosdb_sql_user_defined_function_create_update(client,
+                                      resource_group_name,
+                                      account_name,
+                                      database_name,
+                                      container_name,
+                                      user_defined_function_name,
+                                      user_defined_function_body):
+
+    """Creates or Updates an Azure Cosmos DB SQL user defined function """
+    sql_user_defined_function_resource = SqlUserDefinedFunctionResource(id=user_defined_function_name)
+    sql_user_defined_function_resource.body = user_defined_function_body
+
+    sql_user_defined_function_create_update_resource = SqlUserDefinedFunctionCreateUpdateParameters(
+        resource=sql_user_defined_function_resource,
+        options={})
+
+    return client.create_update_sql_user_defined_function(resource_group_name,
+                                              account_name,
+                                              database_name,
+                                              container_name,
+                                              user_defined_function_name,
+                                              sql_user_defined_function_create_update_resource)
+
+def _populate_sql_trigger_definition(sql_trigger_resource,
+                                       trigger_body,
+                                       trigger_operation,
+                                       trigger_type):
+    if all(arg is None for arg in
+           [trigger_body, trigger_operation, trigger_type]):
+        return False
+
+    if trigger_body is not None:
+        sql_trigger_resource.body = trigger_body
+
+    if trigger_body is not None:
+        sql_trigger_resource.trigger_operation = trigger_operation
+
+    if trigger_body is not None:
```

Comment:
>trigger_body [](start = 7, length = 12)

trigger_type

Reply:
Thanks for pointing this out. Changed it.